cpp/src/streaming/cudf/shuffler.cpp

```diff
@@ -10,37 +10,147 @@
 #include <rmm/cuda_stream_view.hpp>
 
 #include <rapidsmpf/cuda_event.hpp>
+#include <rapidsmpf/cuda_stream.hpp>
 #include <rapidsmpf/streaming/cudf/shuffler.hpp>
 
-namespace rapidsmpf::streaming::node {
+namespace rapidsmpf::streaming {
 
 namespace {
 
 /**
- * @brief Make @p primary wait until all work currently enqueued on @p secondary
- * completes.
+ * @brief Caller side corouting that inserts a partition ID into a set and notifies all
+ * waiting tasks.
  *
- * Records @p event on @p secondary and inserts a wait for that event on @p primary.
- * This is fully asynchronous with respect to the host thread; no host-side blocking.
+ * Wrapping mtx lock and cv notify in a coroutine task to avoid calling coro::sync_wait
+ * multiple times.
```

Comment:
This feels like a comment on the implementation, rather than documentation on how to use the function.